src/box/box.cc

```diff
@@ -4310,11 +4310,22 @@ box_process_fetch_snapshot(struct iostream *io,
 			  "wal_mode = 'none'");
 	}
 
-	say_info("sending current read-view to replica at %s", sio_socketname(io->fd));
+	say_info("sending read-view to replica at %s", sio_socketname(io->fd));
+	/* Used for checkpoint initial join. */
+	struct checkpoint_cursor cursor;
+	struct checkpoint_cursor *cursor_ptr = NULL;
+	if (req.is_checkpoint_join) {
+		memset(&cursor, 0, sizeof(cursor));
+		cursor.vclock = &req.checkpoint_vclock;
+		cursor.start_lsn = req.checkpoint_lsn;
+		cursor.lsn_counter = 0;
```

Comment:
This isn't needed after you did `memset` above. It is already 0.

Reply:
`lsn_counter` initialization was dropped